src/components/Book.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+
+function BookList() {
+  const [books, setBooks] = useState([]);
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const newBook = { title, author };
+    setBooks([...books, newBook]);
+    setTitle('');
+    setAuthor('');
+  };
+
+  return (
+    <div>
+      <h2>Book List</h2>
+      <ul>
+        {books.map((book) => (
+          <li key={null}>
+            {book.title}
+            by
+            {book.author}
+          </li>
+        ))}
+      </ul>
+
+      <h2>Add a Book</h2>
+      <form onSubmit={handleSubmit}>
+        <div>
+          <input id="title" type="text" value={title} onChange={(event) => setTitle(event.target.value)} />
+        </div>
+        <div>
+          <input id="author" type="text" value={author} onChange={(event) => setAuthor(event.target.value)} />
+        </div>
+        <button type="submit">Add new book</button>
+      </form>
```

Comment:
Nice form 👍🏽 but the code would be cleaner and easier to manage if you extracted this into a component (plus it's required) 😉 consider creating a file named `NewBookForm, it should be a component on its own that gets imported into the `BookList` component (yours is called Book for now). As a whole with both components extracted, you should have a structure like below 👇🏽 

```
    <div>
      <h2>Book List</h2>
      <ul>
        {books.map((book) => (
         <Book key={null} title={book.title} author={book.author} />
        ))}
      </ul>
      <h2>Add a Book</h2>
      <NewBookForm />
    </div>
```